[Automation] javac compile fails for org.apache.maven.plugins:maven-jar-plugin:3.4.0

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console
Parsed versions: 3.4.0 3.4.1 3.4.2 3.5.0 4.0.0-beta-1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.4.0 at 2026-06-09T01:04:05Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-jar-plugin:3.3.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-jar-plugin:3.3.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.0 at 2026-06-09T01:04:20Z with exit code 1 after 15s
❌ ERROR: 'native-image run' for 3.4.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:20Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 907ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:23Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.4.0 failed with exit code 1.
FAILED[javac compile][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console
Parsed versions: 3.4.0 3.4.1 3.4.2 3.5.0 4.0.0-beta-1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.4.0 at 2026-06-09T01:04:13Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-jar-plugin:3.3.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^

  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-jar-plugin:3.3.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.0 at 2026-06-09T01:04:27Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 3.4.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:27Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 888ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html
        HelpMojo mojo = new HelpMojo();

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s
10 actionable tasks: 1 executed, 9 up-to-date

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:31Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 3.4.0 failed with exit code 1.
FAILED[javac compile][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/27176806241
Last 300 lines of the log:
```console
Parsed versions: 3.4.0 3.4.1 3.4.2 3.5.0 4.0.0-beta-1
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0
========================================================================================
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage native-image run
========================================================================================
Starting stage 'native-image run' for 3.4.0 at 2026-06-09T01:03:54Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 6s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.apache.maven.plugins:maven-jar-plugin:3.3.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources NO-SOURCE
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol

> Task :compileTestJava FAILED
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();

                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 5 executed, 8 up-to-date

BUILD FAILED in 7s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.apache.maven.plugins:maven-jar-plugin:3.3.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.4.0 at 2026-06-09T01:04:09Z with exit code 1 after 15s
❌ ERROR: 'native-image run' for 3.4.0 failed with exit code 1.
SOFT-FAIL[native-image run][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
========================================================================================
 org.apache.maven.plugins:maven-jar-plugin:3.3.0:3.4.0 stage javac compile
========================================================================================
Starting stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:09Z
Command: env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 856ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE

> Task :compileTestJava/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
import org.apache.maven.plugins.jar.HelpMojo;
                                   ^
  symbol:   class HelpMojo
  location: package org.apache.maven.plugins.jar
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
        ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
        HelpMojo mojo = new HelpMojo();
                            ^
  symbol:   class HelpMojo
  location: class HelpMojoTest
3 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:11: error: cannot find symbol
  import org.apache.maven.plugins.jar.HelpMojo;
                                     ^
    symbol:   class HelpMojo
    location: package org.apache.maven.plugins.jar
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
          ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java:17: error: cannot find symbol
          HelpMojo mojo = new HelpMojo();
                              ^
    symbol:   class HelpMojo
    location: class HelpMojoTest
  3 errors

* Try:
> Check your code and dependencies to fix the compilation error(s)
 FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.3.0/build/reports/problems/problems-report.html

> Run with --scan to generate a Build Scan (Powered by Develocity).
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

BUILD FAILED in 897ms

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 3.4.0 at 2026-06-09T01:04:12Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 3.4.0 failed with exit code 1.
FAILED[javac compile][3.4.0][env GVM_TCK_LV=3.4.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.apache.maven.plugins:maven-jar-plugin:3.3.0]
```

--- a/metadata/org.apache.maven.plugins/maven-jar-plugin/3.4.0/reachability-metadata.json
+++ b/metadata/org.apache.maven.plugins/maven-jar-plugin/3.4.0/reachability-metadata.json
@@ -1,0 +1,42 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "type": "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "type": "short[]"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "type": "short[][]"
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "glob": "META-INF/maven/org.apache.maven.plugins/maven-jar-plugin/plugin-help.xml"
+    },
+    {
+      "condition": {
+        "typeReached": "org.apache.maven.plugins.maven_jar_plugin.HelpMojo"
+      },
+      "glob": "META-INF/services/javax.xml.parsers.DocumentBuilderFactory"
+    }
+  ]
+}

--- a/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
@@ -11,7 +11,7 @@
       "3.4.0"
     ],
     "allowed-packages" : [
-      "org.apache.maven.plugins.jar"
+      "org.apache.maven.plugins.maven_jar_plugin"
     ]
   },
   {

--- a/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "3.4.0",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-jar-plugin/$version$/maven-jar-plugin-$version$-sources.jar",
+    "repository-url" : "https://github.com/apache/maven-jar-plugin",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-jar-plugin/$version$/maven-jar-plugin-$version$-source-release.zip",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-jar-plugin/$version$/maven-jar-plugin-$version$-javadoc.jar",
+    "description" : "Apache Maven JAR Plugin builds Java Archive (JAR) files from a Maven project's compiled classes and resources. It also provides goals for packaging the main artifact and attaching a test JAR containing test classes.",
     "tested-versions" : [
       "3.4.0"
     ],

--- a/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
+++ b/metadata/org.apache.maven.plugins/maven-jar-plugin/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "3.4.0",
+    "tested-versions" : [
+      "3.4.0"
+    ],
+    "allowed-packages" : [
+      "org.apache.maven.plugins.jar"
+    ]
+  },
+  {
     "metadata-version" : "3.3.0",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-jar-plugin/$version$/maven-jar-plugin-$version$-sources.jar",
     "repository-url" : "https://github.com/apache/maven-jar-plugin",

--- a/stats/org.apache.maven.plugins/maven-jar-plugin/3.4.0/execution-metrics.json
+++ b/stats/org.apache.maven.plugins/maven-jar-plugin/3.4.0/execution-metrics.json
@@ -1,0 +1,131 @@
+{
+  "fix_javac_fail:2026-06-09": {
+    "timestamp": "2026-06-09T07:48:12.558577Z",
+    "library": "org.apache.maven.plugins:maven-jar-plugin:3.4.0",
+    "starting_commit": "726bc13f0674109f239787c6d30069912b02b895",
+    "ending_commit": "eb54d82513bdeecaa09a7906daaa61209355f2b1",
+    "previous_library": "org.apache.maven.plugins:maven-jar-plugin:3.3.0",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "3.4.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 538,
+          "missed": 922,
+          "ratio": 0.368493,
+          "total": 1460
+        },
+        "line": {
+          "covered": 103,
+          "missed": 175,
+          "ratio": 0.370504,
+          "total": 278
+        },
+        "method": {
+          "covered": 14,
+          "missed": 33,
+          "ratio": 0.297872,
+          "total": 47
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "3.3.0",
+      "dynamicAccess": {
+        "breakdown": {
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 1,
+            "totalCalls": 1
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 1,
+        "totalCalls": 1
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 538,
+          "missed": 671,
+          "ratio": 0.444996,
+          "total": 1209
+        },
+        "line": {
+          "covered": 102,
+          "missed": 125,
+          "ratio": 0.449339,
+          "total": 227
+        },
+        "method": {
+          "covered": 14,
+          "missed": 21,
+          "ratio": 0.4,
+          "total": 35
+        }
+      }
+    },
+    "library_preparation_preflight": {
+      "status": "completed",
+      "action": "no_action",
+      "issue_number": 8227,
+      "issue_label": "fails-javac-compile",
+      "library": "org.apache.maven.plugins:maven-jar-plugin:3.4.0",
+      "summary": "No extra deterministic setup is needed. The 3.4.0 artifact contains the plugin implementation and its normal transitive runtime dependencies; the javac failure is caused by the generated HelpMojo class moving from org.apache.maven.plugins.jar.HelpMojo in 3.3.0 to org.apache.maven.plugins.maven_jar_plugin.HelpMojo in 3.4.0, not by a missing dependency, service, or initialization step.",
+      "deterministic_setup": [],
+      "agent_guidance": "Do not assume the generated help mojo stays in org.apache.maven.plugins.jar; for 3.4.0 use org.apache.maven.plugins.maven_jar_plugin.HelpMojo or exercise stable plugin classes such as org.apache.maven.plugins.jar.JarMojo/TestJarMojo with local temporary files only if additional coverage is needed.",
+      "risks": [
+        "Directly executing the real jar/test-jar mojos is more involved than executing the help mojo because Maven normally injects project/session/output-directory fields; avoid adding broad Maven harness setup unless the generated test intentionally covers archive creation."
+      ],
+      "applied_setup": [],
+      "input_evidence": [
+        {
+          "name": "issue",
+          "summary": "#8227 [Automation] javac compile fails for org.apache.maven.plugins:maven-jar-plugin:3.4.0; label=fails-javac-compile; body_chars=8015"
+        },
+        {
+          "name": "existing_tests",
+          "summary": "6 existing test file path(s) included"
+        }
+      ],
+      "current_library": "org.apache.maven.plugins:maven-jar-plugin:3.3.0",
+      "new_version": "3.4.0",
+      "model": "oca/gpt-5.5",
+      "input_tokens_used": 25372,
+      "output_tokens_used": 2755,
+      "prompt_path": "library-preflight-prompt.txt",
+      "raw_response_path": "library-preflight-response.txt",
+      "session_log_path": "/home/vjovanov/c/forge/forge1/graalvm-reachability-metadata/forge/logs/org.apache.maven.plugins:maven-jar-plugin:3.4.0/library-preparation-preflight/pi-generation-2026-06-09T07-38-15-865991Z.log"
+    },
+    "metrics": {
+      "input_tokens_used": 10899,
+      "cached_input_tokens_used": 90112,
+      "output_tokens_used": 900,
+      "iterations": 1,
+      "cost_usd": 0.0721,
+      "tested_library_loc": 103,
+      "metadata_entries": 6,
+      "previous_library_metadata_entries": 6,
+      "code_coverage_percent": 37.05,
+      "previous_library_coverage_percent": 44.93
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java",
+      "metadata_file": "metadata/org.apache.maven.plugins/maven-jar-plugin/3.4.0/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.apache.maven.plugins/maven-jar-plugin/3.4.0/stats.json
+++ b/stats/org.apache.maven.plugins/maven-jar-plugin/3.4.0/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "3.4.0",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 1,
+          "totalCalls" : 1
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 1,
+      "totalCalls" : 1
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 538,
+        "missed" : 922,
+        "ratio" : 0.368493,
+        "total" : 1460
+      },
+      "line" : {
+        "covered" : 103,
+        "missed" : 175,
+        "ratio" : 0.370504,
+        "total" : 278
+      },
+      "method" : {
+        "covered" : 14,
+        "missed" : 33,
+        "ratio" : 0.297872,
+        "total" : 47
+      }
+    }
+  } ]
+}

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/.gitignore
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/build.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/build.gradle
@@ -1,0 +1,30 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation("org.apache.maven.plugins:maven-jar-plugin:$libraryVersion") {
+        exclude group: "org.sonatype.sisu", module: "sisu-guice"
+    }
+    testImplementation 'org.apache.maven:maven-plugin-api:3.2.5'
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/gradle.properties
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.apache.maven.plugins:maven-jar-plugin:3.4.0
+library.version = 3.4.0
+metadata.dir = org.apache.maven.plugins/maven-jar-plugin/3.4.0/

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/settings.gradle
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.apache.maven.plugins.maven-jar-plugin_tests'

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java
@@ -8,7 +8,7 @@ package org_apache_maven_plugins.maven_jar_plugin;
 
 import static org.assertj.core.api.Assertions.assertThatCode;
 
-import org.apache.maven.plugins.jar.HelpMojo;
+import org.apache.maven.plugins.maven_jar_plugin.HelpMojo;
 import org.junit.jupiter.api.Test;
 
 public class HelpMojoTest {

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/src/test/java/org_apache_maven_plugins/maven_jar_plugin/HelpMojoTest.java
@@ -1,0 +1,21 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_apache_maven_plugins.maven_jar_plugin;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import org.apache.maven.plugins.jar.HelpMojo;
+import org.junit.jupiter.api.Test;
+
+public class HelpMojoTest {
+    @Test
+    void executeLoadsPluginHelpResource() {
+        HelpMojo mojo = new HelpMojo();
+
+        assertThatCode(mojo::execute).doesNotThrowAnyException();
+    }
+}

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/user-code-filter.json
@@ -7,6 +7,9 @@
       "includeClasses" : "org.apache.maven.plugins.jar.**"
     },
     {
+      "includeClasses" : "org.apache.maven.plugins.**"
+    },
+    {
       "includeClasses" : "org_apache_maven_plugins.maven_jar_plugin.**"
     }
   ]

--- a/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/user-code-filter.json
+++ b/tests/src/org.apache.maven.plugins/maven-jar-plugin/3.4.0/user-code-filter.json
@@ -1,0 +1,13 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.apache.maven.plugins.jar.**"
+    },
+    {
+      "includeClasses" : "org_apache_maven_plugins.maven_jar_plugin.**"
+    }
+  ]
+}

